crates/runtime/src/builder.rs

```diff
@@ -206,9 +206,10 @@ impl RuntimeBuilder {
                 eprintln!("Failed to initialize extension {extension_name}: {err}");
             } else {
                 extensions.insert(extension_name.into(), extension.into());
-            };
+            }
         }
         rt.extensions = Arc::new(RwLock::new(extensions));
+        tools::factory::register_all_factories(Arc::new(rt.clone())).await;
```

Comment:
Let's follow the model that the `accelerator_engine_registry` does by pre-creating the registry, and passing only the registry into `tools::factory::register_all_factories` instead of passing the entire runtime object. That also obviates the need to clone the runtime object itself, which is a bit weird.

Reply:
This is awkward but the `Arc<Runtime>` is required as catalogs need a reference back to the `Runtime` so that it can make tools when needed. The alternative is to change how the tool catalogs work.